metals/src/main/scala/scala/meta/internal/metals/CodeActionProvider.scala

```diff
@@ -0,0 +1,28 @@
+package scala.meta.internal.metals
+
+import org.eclipse.{lsp4j => l}
+import scala.meta.pc.CancelToken
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+final class CodeActionProvider(
+    compilers: Compilers
+) {
+
+  def codeActions(
+      params: l.CodeActionParams,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = {
+
+    for {
+      quickFixes <- Future
+        .sequence(
+          List(QuickFix.ImportMissingSymbol)
```

Comment:
this will grow into a list of all quickfixes, e.g.

```scala
List(QuickFix.ImportMissingSymbol, QuickFix.ImplementAbstractMembers, QuickFix.RemoveUnusedImport)
```